app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -46,7 +52,7 @@ const CONFIRMATION_METHODS = Object.freeze([
  * @returns PPOMMiddleware function.
  */
 export function createPPOMMiddleware<
-  Params extends JsonRpcParams,
+  Params extends (string | { to: string })[],
```

Comment:
Narrowed generic constraint based on the expected shape of the request params object outlined in the function body.